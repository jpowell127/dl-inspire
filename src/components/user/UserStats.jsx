import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import { Statistic } from "semantic-ui-react";

function UserStats({ stats }) {
  const { followers, following, appreciations, views, comments } = stats;

  return (
    <Statistic.Group size="mini" widths="five">
      <Statistic color="red">
        <Statistic.Value>{followers}</Statistic.Value>
        <Statistic.Label size="mini">Followers</Statistic.Label>
      </Statistic>
      <Statistic color="orange">
        <Statistic.Value>{following}</Statistic.Value>
        <Statistic.Label>Following</Statistic.Label>
      </Statistic>
      <Statistic color="yellow">
        <Statistic.Value>{appreciations}</Statistic.Value>
        <Statistic.Label>Appreciations</Statistic.Label>
      </Statistic>
      <Statistic color="green">
        <Statistic.Value>{views}</Statistic.Value>
        <Statistic.Label>Views</Statistic.Label>
      </Statistic>
      <Statistic color="teal">
        <Statistic.Value>{comments}</Statistic.Value>
        <Statistic.Label>Comments</Statistic.Label>
      </Statistic>
    </Statistic.Group>
  );
}

UserStats.propTypes = {
  stats: PropTypes.shape({
    followers: PropTypes.number.isRequired,
    following: PropTypes.number.isRequired,
    appreciations: PropTypes.number.isRequired,
    views: PropTypes.number.isRequired,
    comments: PropTypes.number.isRequired
  }).isRequired
};

export default UserStats;
