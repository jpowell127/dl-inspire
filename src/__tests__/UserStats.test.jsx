import React from "react";
import UserStats from "../components/user/UserStats";
import MockStats from "./mocks/mock.stats";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

describe("<UserStats />", () => {
  it("renders correctly", () => {
    const wrapper = mount(<UserStats stats={MockStats} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
  it("renders the correct stats", () => {
    const wrapper = mount(<UserStats stats={MockStats} />);
    expect(wrapper.props().stats).toBe(MockStats);
    wrapper.unmount();
  });
});
