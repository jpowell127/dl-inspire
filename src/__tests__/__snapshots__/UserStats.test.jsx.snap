// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserStats /> renders correctly 1`] = `
<UserStats
  stats={
    Object {
      "appreciations": 1,
      "comments": 1,
      "followers": 1,
      "following": 1,
      "views": 1,
    }
  }
>
  <StatisticGroup
    size="mini"
    widths="five"
  >
    <div
      className="ui mini five statistics"
    >
      <Statistic
        color="red"
      >
        <div
          className="ui red statistic"
        >
          <StatisticValue>
            <div
              className="value"
            >
              1
            </div>
          </StatisticValue>
          <StatisticLabel>
            <div
              className="label"
            >
              Followers
            </div>
          </StatisticLabel>
        </div>
      </Statistic>
      <Statistic
        color="orange"
      >
        <div
          className="ui orange statistic"
        >
          <StatisticValue>
            <div
              className="value"
            >
              1
            </div>
          </StatisticValue>
          <StatisticLabel>
            <div
              className="label"
            >
              Following
            </div>
          </StatisticLabel>
        </div>
      </Statistic>
      <Statistic
        color="yellow"
      >
        <div
          className="ui yellow statistic"
        >
          <StatisticValue>
            <div
              className="value"
            >
              1
            </div>
          </StatisticValue>
          <StatisticLabel>
            <div
              className="label"
            >
              Appreciations
            </div>
          </StatisticLabel>
        </div>
      </Statistic>
      <Statistic
        color="green"
      >
        <div
          className="ui green statistic"
        >
          <StatisticValue>
            <div
              className="value"
            >
              1
            </div>
          </StatisticValue>
          <StatisticLabel>
            <div
              className="label"
            >
              Views
            </div>
          </StatisticLabel>
        </div>
      </Statistic>
      <Statistic
        color="teal"
      >
        <div
          className="ui teal statistic"
        >
          <StatisticValue>
            <div
              className="value"
            >
              1
            </div>
          </StatisticValue>
          <StatisticLabel>
            <div
              className="label"
            >
              Comments
            </div>
          </StatisticLabel>
        </div>
      </Statistic>
    </div>
  </StatisticGroup>
</UserStats>
`;
